invenio_oauthclient/receivers.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+from invenio_base.signals import secret_key_changed
+from invenio_oauthclient.models import RemoteToken
+from invenio_base.utils import rebuild_encrypted_field
+
+
+@secret_key_changed.connect
```

Comment:
We had issues in the past with side effects of importing files. As this code is optional it would be better to call explicitly `secret_key_changed.connect(secret_key_changed.connect)` instead of using it as a decorator. Also this file should be in a `contrib` directory as it depends on `invenio_base`.